examples/CSVHistorian/csv_historian/historian.py

```diff
@@ -0,0 +1,142 @@
+# agency of the United States Government.  Neither the United States
+# Government nor the United States Department of Energy, nor Battelle,
+# nor any of their employees, nor any jurisdiction or organization
+# that has cooperated in the development of these materials, makes
+# any warranty, express or implied, or assumes any legal liability
+# or responsibility for the accuracy, completeness, or usefulness or
+# any information, apparatus, product, software, or process disclosed,
+# or represents that its use would not infringe privately owned rights.
+#
+# Reference herein to any specific commercial product, process, or
+# service by trade name, trademark, manufacturer, or otherwise does
+# not necessarily constitute or imply its endorsement, recommendation,
+# r favoring by the United States Government or any agency thereof,
+# or Battelle Memorial Institute. The views and opinions of authors
+# expressed herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY
+# operated by BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+
+# }}}
+from __future__ import absolute_import, print_function
+
+import sys
+import logging
+
+from volttron.platform.agent import utils
+from volttron.platform.agent.base_historian import BaseHistorian
+
+import csv
+
+
+utils.setup_logging()
+_log = logging.getLogger(__name__)
+__version__ = '1.0.1'
+
+
+def historian(config_path, **kwargs):
+    """
+    This method is called by the :py:func:`crate_historian.historian.main` to parse
+    the passed config file or configuration dictionary object, validate the
+    configuration entries, and create an instance of CSVHistorian
+
+    :param config_path: could be a path to a configuration file or can be a
+                        dictionary object
+    :param kwargs: additional keyword arguments if any
+    :return: an instance of :py:class:`CSVHistorian`
+    """
+    if isinstance(config_path, dict):
+        config_dict = config_path
+    else:
+        config_dict = utils.load_config(config_path)
+
+    output_path = config_dict.get("output", "~/historian_output.csv")
+
+    return CSVHistorian(output_path = output_path, **kwargs)
+
+
+class CSVHistorian(BaseHistorian):
+    """
+    Historian that stores the data into crate tables.
+
+    """
+
+    def __init__(self, output_path="", **kwargs):
+        self.output_path = output_path
+        self.csv_dict = None
+        super(CSVHistorian, self).__init__(**kwargs)
+
+
+    def publish_to_historian(self, to_publish_list):
+        for record in to_publish_list:
+            row = {}
+            row["timestamp"] = record["timestamp"]
+
+            row["source"] = record["source"]
+            row["topic"] = record["topic"]
+            row["value"] = record["value"]
+
+            self.csv_dict.writerow(row)
+
+        self.report_all_handled()
+        self.f.flush()
+
```

Comment:
please add stubs for the methods that a "real" historian would need to implement. i.e a query_historian function.